src/test/field-element-creation-strategy.test.ts

```diff
@@ -292,3 +295,55 @@ test('create div show password checkbox with label', () => {
     // then
     expect(element.outerHTML).toEqual(expectedHTMLLabel);
 })
+
+describe('fetchCountryPhoneCodes', () => {
+    global.fetch = jest.fn();
+
+    process.env.API_KEY = 'mockedAPIUrl';
+
+    const mockApiResponse = {
+        US: { country_name: 'United States', dialling_code: '+1' },
+        GB: { country_name: 'United Kingdom', dialling_code: '+44' }
+    };
+    beforeEach(() => {
+        (fetch as jest.Mock).mockClear();
+    });
+    
+    it('should fetch country phone codes and return a Map', async () => {
+        (fetch as jest.Mock) .mockResolvedValueOnce({
+            json: () => Promise.resolve(mockApiResponse)
+        });
+        const expectedMap = new Map([
+            ['United States', '+1'],
+            ['United Kingdom', '+44']
+        ]);
+        const result = await new FieldElementCreationStrategy().fetchCountryPhoneCodes();
+        expect(result).toEqual(expectedMap);
+        expect(fetch).toHaveBeenCalledTimes(1);
+        expect(fetch).toHaveBeenCalledWith(process.env.API_KEY);
+    });
+
+    it('create country options in select element', async () => {
+        const select = selectCreator;
+        (fetch as jest.Mock).mockResolvedValueOnce({
+            json: () => Promise.resolve(mockApiResponse)
+        });
+        const fieldElementCreationStrategy = new FieldElementCreationStrategy();
+        const createCountryOptions = fieldElementCreationStrategy.createCountryOptions.bind(fieldElementCreationStrategy);
+        await createCountryOptions(select);
+        expect(select.children.length).toBe(1);
+        expect(select.children[0].textContent).toBe('United States');
```

Comment:
mozesz sobie to select.children[0] wydzielic do zmiennej bo korzystasz z tego 2x to samo z elementem na indeksie 1

Reply:
Btw. bardzo ladnie to ogarnelas :) jestem zadowolony